### PR TITLE
Fix package name in file menu

### PR DESCRIPTION
**Pull request by Slaviiiii**

### CODE REVIEW

File: dev-server-api/routes/api/package_info/create.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  }),
+  jsonResponse: z.object({
+    package_info: z.object({
+      name: z.string()
+    })
+  }),
+  auth: "none",
+})(async (req, ctx) => {
+  const package_name = req.jsonBody.package_name
+  console.log("PACKAGE NAME", package_name)
```
Comment (seveibar) [inline dev-server-api/routes/api/package_info/create.ts]:
```suggestion
```

File: dev-server-api/routes/api/package_info/get.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+})(async (req, ctx) => {
+  console.log("in endpoint")
+  const db_package_info_request: any = await ctx.db
+    .selectFrom("package_info")
+    .select("name")
+    .execute()
+
+  return ctx.json({
+    package_info: db_package_info_request[0]
+  })
```
Comment (seveibar) [inline dev-server-api/routes/api/package_info/get.ts]:
```suggestion
  const package_info = await ctx.db
    .selectFrom("package_info")
    .select("name")
    .executeTakeFirstOrThrow()

  return ctx.json({  package_info  })
```

File: dev-server-api/src/db/get-db.ts [hunk 3]
```diff
@@ -98,12 +103,15 @@ export const getDb = async (): Promise<Kysely<KyselyDatabaseSchema>> => {
...
     FROM sqlite_master
-    WHERE type='table' AND name='dev_package_example'
+    WHERE type='table' AND name IN ('dev_package_example', 'export_request', 'export_file', 'package_info')
   `.execute(db)
 
-  if (schemaExistsResult.rows.length === 0) {
+  // Check if the number of existing tables matches the number of required tables
+  if (schemaExistsResult.rows.length < 4) {
     await createSchema(db)
   }
```
Comment (seveibar) [inline dev-server-api/src/db/get-db.ts:112]: let's remove the changes to this file, this is called "migration management" and it has a very specific solution. If you have issues where your db is out of date just remove it for now (delete the `.tscircuit` directory). The underlying issue is on windows we currently aren't reseting the database

Comment (seveibar) [reply dev-server-api/src/db/get-db.ts:112]: to reset this file, you can do something like `git checkout origin/main -- path/to-file`

File: dev-server-frontend/src/ExampleContentView.tsx [hunk 4]
```diff
@@ -141,28 +141,28 @@ export const ExampleContentView = () => {
...
+          <div className="p-4 m-16 whitespace-pre border border-gray-200 rounded-lg shadow-lg bg-gray-50">
             Loading...
           </div>
         </div>
       )}
       {pkg && pkg.is_loading && (
-        <div className="absolute top-0 right-0 bg-white p-4 py-2 m-4 rounded-md flex items-center z-10 shadow-lg border border-gray-200">
-          <div className="border-2 border-blue-400 border-t-transparent rounded-full w-4 h-4 animate-spin mr-2"></div>
+        <div className="absolute top-0 right-0 z-10 flex items-center p-4 py-2 m-4 bg-white border border-gray-200 rounded-md shadow-lg">
+          <div className="w-4 h-4 mr-2 border-2 border-blue-400 rounded-full border-t-transparent animate-spin"></div>
```
Comment (seveibar) [inline dev-server-frontend/src/ExampleContentView.tsx:165]: what do these changes do? Can you either describe them & provide a screenshot or revert them? Make sure to include any visual changes with a screenshot in the PR description

**Review by seveibar — CHANGES_REQUESTED** [review]
looking good! see comments

File: lib/cmd-fns/dev/index.ts [hunk 5]
```diff
@@ -71,6 +72,19 @@ export const devCmd = async (ctx: AppContext, args: any) => {
 
   // Reset the database, allows migration to re-run
   await devServerAxios.post("/api/dev_server/reset")
+    .catch(e => {
+      console.log("Failed to reset database, continuing anyway...")
```
Comment (seveibar) [inline lib/cmd-fns/dev/index.ts:76]: I'm going to create an issue from this comment to note we need to fix this on windows

File: dev-server-frontend/src/ExampleContentView.tsx (hunk 4 above)
Comment (Slaviiiii) [reply dev-server-frontend/src/ExampleContentView.tsx:165]: These changes are just formatting the styles

Comment (seveibar) [reply dev-server-frontend/src/ExampleContentView.tsx:165]: ahh ok

**Review by seveibar — APPROVED** [review]

**Pull request merged**